intrinsic_cal/src/rail_cal.cpp

```diff
@@ -191,32 +213,38 @@ bool RailCalService::executeCallBack( intrinsic_cal::rail_ical_run::Request &req
   industrial_extrinsic_cal::Cost_function cost_type = industrial_extrinsic_cal::cost_functions::CameraReprjErrorWithDistortion;
   Problem problem; // note, a new problem gets created each time execute is called, so old observation data is not re-used.
 
-  // set initial conditions, 
+  // set initial conditions,
   // x of target aligned with y of rail    [ 0  0  1 ]
   // y of target aligned with z of rail    [ 1  0  0 ]
   // z of target aligned with x of rail,   [ 0  1  0 ]
   // target about D0 meters behind rail
-  tf::Matrix3x3 R; 
-  R[0][0] = 1;   R[0][1] = 0;   R[0][2] = 0;
-  R[1][0] = 0;   R[1][1] = -1;   R[1][2] = 0;
+  tf::Matrix3x3 R;
+  //R[0][0] = 1;   R[0][1] = 0;   R[0][2] = 0;
+  //R[1][0] = 0;   R[1][1] = -1;   R[1][2] = 0;
+  //R[2][0] = 0;   R[2][1] = 0;   R[2][2] = -1;
+  R[0][0] = 0;   R[0][1] = -1.0;   R[0][2] = 0;
+  R[1][0] = -1.0;   R[1][1] = 0;   R[1][2] = 0;
   R[2][0] = 0;   R[2][1] = 0;   R[2][2] = -1;
-  
-  target_->pose_.setBasis(R);
-  target_->pose_.setOrigin(-0.011, 0.05, D0_);
-  target_->pose_.show("initial target pose");  
+
+  //target_->pose_.setBasis(R);
+  // Need to use setQuaternion because setBasis does not work for some rotations (results in flipped yaw values and negative focal lengths)
```

Comment:
Is there something wrong with setBasis(), or are the results just mathematically ambiguous and you are trying to remove the ambiguity?